### DISCUSSION TITLE
Using other WSGI server

### DISCUSSION THREAD

**Discussion by EdificomSA** [nautobot/nautobot#6094] [Ideas]
Has anyone deployed Nautobot with another WSGI server like [granian](https://github.com/emmett-framework/granian) or [socketify.py](https://github.com/cirospaciari/socketify.py) ? Any issues/troubles ?

uWSGI replaced Gunicorn in 2021 as default in Nautobot
I [see ](https://github.com/nautobot/nautobot/blob/develop/nautobot/core/management/commands/start.py)the nautobot-server start command uses [django_webserver](https://github.com/lincolnloop/django-webserver) pyuwsgi to run the server
django_webserver also supports Gunicorn, uvicorn and waitress but was not updated since 2020

I'm thinking, would it make sense to add a switch (e.g. env var) and optional deps to choose from these WSGI servers or even "custom" ones ? I can "override" by running wsgi server directly instead of using nautobot-server
Documentation doesn't tell much about using other WSGI (which is fine as it works well with uWSGI) except that it should work

**Comment by brettplarson** [comment]
I asked on Slack and someone from NTC said there are no current plans for this. 

> the thinking is it's a) in a good place, and b) would require breaking changes (read: a major release). so it'll be in the mix for 4.0, but there's no date set for that yet, and i can say confidently there won't be a 4.0 until at least 2H 27
